pkg/collector/python/init.go

```diff
@@ -303,31 +303,34 @@ func pathToBinary(name string, ignoreErrors bool) (string, error) {
 }
 
 func resolvePythonExecPath(ignoreErrors bool) (string, error) {
-	// Since the install location can be set by the user on Windows we use relative import
-	if runtime.GOOS == "windows" {
-		_here, err := executable.Folder()
+	// Allow to relatively import python
+	_here, err := executable.Folder()
+	if err != nil {
+		log.Warnf("Error getting executable folder: %v", err)
+		log.Warnf("Trying again allowing symlink resolution to fail")
+		_here, err = executable.FolderAllowSymlinkFailure()
 		if err != nil {
-			log.Warnf("Error getting executable folder: %v", err)
-			log.Warnf("Trying again allowing symlink resolution to fail")
-			_here, err = executable.FolderAllowSymlinkFailure()
-			if err != nil {
-				log.Warnf("Error getting executable folder w/o symlinks: %v", err)
-			}
-		}
-		log.Debugf("Executable folder is %v", _here)
-
-		embeddedPythonHome3 := filepath.Join(_here, "..", "embedded3")
-
-		// We want to use the path-relative embedded2/3 directories above by default.
-		// They will be correct for normal installation on Windows. However, if they
-		// are not present for cases like running unit tests, fall back to the compile
-		// time values.
-		if _, err := os.Stat(embeddedPythonHome3); os.IsNotExist(err) {
-			log.Warnf("Relative embedded directory not found for Python 3. Using default: %s", pythonHome3)
-		} else {
-			pythonHome3 = embeddedPythonHome3
+			log.Warnf("Error getting executable folder w/o symlinks: %v", err)
 		}
 	}
+	log.Debugf("Executable folder is %v", _here)
+
+	var embeddedPythonHome3 string
+	if runtime.GOOS == "windows" {
+		embeddedPythonHome3 = filepath.Join(_here, "..", "embedded3")
+	} else { // IIRC both macOS and Linux would have the same relative paths so it makes sense as the "default" value
```

Comment:
Nit: the comment in my suggestion was just explaining my reasoning for the suggestion, so I'd tweak it a bit before merge.

```suggestion
	} else { // Both macOS and Linux have the same relative paths
```